### PR TITLE
i created new auth and added filter in BookViewSet('is archived')

### PR DESCRIPTION
**Pull request by m1rageg**

### CODE REVIEW

File: books/viewsets.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 
 class BookViewSet(ModelViewSet):
-    queryset = Book.objects.select_related('country').prefetch_related('authors')
+    queryset = Book.objects.select_related('country').prefetch_related('authors').filter(is_archived=False)
```
Comment (p1v2) [inline books/viewsets.py:10]: good!

File: hillel_django/authentication.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+class GloryToUkraineAuthentication(BaseAuthentication):
+    def authenticate(self, request):
+        query_params = request.GET
+
+        if query_params.get("haslo") == "SlavaUkraini":
+            return User.objects.get(username="admin"), None
+
+
+class SecretHeaderAuthentication(BaseAuthentication):
+    def authenticate(self, request: Request):
```
Comment (p1v2) [inline hillel_django/authentication.py:19]: 2 things here to fix:
1) If you try to do the request without servert-header, then your `secret_header` variable will be None. And you'll have an error in line 17: None type object has no attribute startswith. So your server will return 500 error. Try it:
```
request.post("http://localhost:8000/api/books", headers={}
```

2) If you pass the header but the username is not valid. For example this request:
```
request.post("http://localhost:8000/api/books", headers={"secret-header": "Some super secret invalid_username"}
```
Your request should return 401 but will return 500 in this case.

https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication - check here how to handle invalid username exceptions.

**Review by p1v2 — CHANGES_REQUESTED** [review]
Pls check you authentication classes.

**Comment by m1rageg** [comment]
i corrected the comments

**Pull request closed without merging**